### ISSUE TITLE
[engine] Inconsistent documents of build env on Windows host

### ISSUE THREAD

**Issue by Sunbreak** [flutter/flutter#61440]
**Should `GYP_MSVS_VERSION` be empty, `2017` or `2019`?**

- compiling-for-windows

According to https://github.com/flutter/flutter/wiki/Compiling-the-engine#compiling-for-windows, there no need for `GYP_MSVS_VERSION`

```cmd
DEPOT_TOOLS_WIN_TOOLCHAIN=0
GYP_MSVS_OVERRIDE_PATH="C:\Program Files (x86)/Microsoft Visual Studio/2019/Community" (or your location for Visual Studio)
WINDOWSSDKDIR="C:\Program Files (x86)\Windows Kits\10" (or your location for Windows Kits)
```

- compiling-for-the-web-on-windows

According to https://github.com/flutter/flutter/wiki/Compiling-the-engine#compiling-for-the-web-on-windows, there a version conflict in `GYP_MSVS_OVERRIDE_PATH` and `GYP_MSVS_VERSION`

```cmd
GYP_MSVS_OVERRIDE_PATH = "C:\Program Files (x86)\Microsoft Visual Studio\2019\Community"
GYP_MSVS_VERSION = 2017
```

- buildroot/build/vs_toolchain.py

As to https://github.com/flutter/buildroot/blob/f9331b6af6e690516a2ffdc14622c768977e3c97/build/vs_toolchain.py#L93-L97, `GYP_MSVS_VERSION` could be inferred from `GetVisualStudioVersion()` and left `empty`

```python
  elif sys.platform == 'win32' and not depot_tools_win_toolchain:
    if not 'GYP_MSVS_OVERRIDE_PATH' in os.environ:
      os.environ['GYP_MSVS_OVERRIDE_PATH'] = DetectVisualStudioPath()
    if not 'GYP_MSVS_VERSION' in os.environ:
      os.environ['GYP_MSVS_VERSION'] = GetVisualStudioVersion()
```

**Comment by deakjahn** [comment]
Did you actually manage to compile? I gave up the Windows experiments some time ago and moved to Linux because the Python code has incompatible path handling.

**Comment by Sunbreak** [comment]
> Did you actually manage to compile? I gave up the Windows experiments some time ago and moved to Linux because the Python code has incompatible path handling.

Yep, definitely. We are working on the #557793

**Comment by deakjahn** [comment]
I'd be very much interested in the outcome. This is about Web apps and not desktop one, isn't it? CanvasKit or DomCanvas?

I'll fire up my notebook and try the engine installation again. I remember that I had most of it working, just the last step `python .\flutter\tools\gn --unoptimized` was failing. And I did find a bug right here about it, I'll try to dig that out now.

**Comment by deakjahn** [comment]
This one: https://github.com/flutter/flutter/issues/28720

And I also documented there how I could set it up on Linux under Windows because I found the original description slight out of tune.

**Comment by Sunbreak** [comment]
> I'd be very much interested in the outcome. This is about Web apps and not desktop ones, isn't it? CanvasKit or DomCanvas?
> 
> I'll fire up my notebook and try the engine installation again. I remember that I had most of it working, just the last step `python .\flutter\tools\gn --unoptimized` was failing. And I did find a bug right here about it, I'll try to dig that out now.

#57793 is a Windows engine problem. I build Windows engine on Windows and Web engine on Mac.

**Comment by deakjahn** [comment]
By Windows engine you mean the engine used for a Windows desktop app, not a web app that you just happen to test while running on Windows? Because I had a problem with the latter, so that might mean that my problem is still there...

**Comment by Sunbreak** [comment]
> By Windows engine you mean the engine used for a Windows desktop app, not a web app that you just happen to test while running on Windows? Because I had a problem with the latter, so that might mean that my problem is still there...

Yeah. Most of us work on Mac. We'll investigate into Web engine compilation on Windows someday maybe.

**Comment by deakjahn** [comment]
Well, chances are you'll find the same then that it doesn't work. :-)

**Comment by Sunbreak** [comment]
> Well, chances are you'll find the same then that it doesn't work. :-)

@deakjahn I manage to compile Flutter Web engine on Windows host with `DEPOT_TOOLS_WIN_TOOLCHAIN` and `GYP_MSVS_OVERRIDE_PATH` set

**Comment by deakjahn** [comment]
I wonder how you do it. I can't reach that phase, tried in once more and no cigar. I get the same errors as in the linked bug.

**Comment by Sunbreak** [comment]
> **Should `GYP_MSVS_VERSION` be empty, `2017` or `2019`?**

`empty` if fine. Here is the blog in Chinese https://juejin.cn/post/6854573209262489607

@chinmaygarde Could you help update the wiki?
